raiden/transfer/mediated_transfer/target.py

```diff
@@ -84,12 +84,16 @@ def handle_inittarget(
         # the handler handle_receive_lockedtransfer.
         target_state = TargetTransferState(route, transfer)
 
-        safe_to_wait, unsafe_msg = is_safe_to_wait(
+        safe_to_wait, _ = is_safe_to_wait(
             transfer.lock.expiration,
             channel_state.reveal_timeout,
             block_number,
         )
 
+        # If there is not enough time to safely unlock the lock on-chain
+        # silently let the transfer expire. The target task must be created to
+        # handle the ReceiveLockExpired state change, which will clear the
+        # expired lock.
```

Comment:
I don't get how this comment makes sense here. Where is the target task created it talks about?

Reply:
So perhaps the comment can be reworded a bit but what I believe it means is:

- The target task is created [here](https://github.com/raiden-network/raiden/blob/325f4327583dc8e75677751865e63dda600bd929/raiden/transfer/mediated_transfer/target.py#L85) under the `is_valid` branch.

- Afterwards we check if it's safe to wait, and if it is we send the `SecretRequest`. If it's not we still [return](https://github.com/raiden-network/raiden/blob/325f4327583dc8e75677751865e63dda600bd929/raiden/transfer/mediated_transfer/target.py#L118) the target task.

- So then the created target task can [handle the expired lock](https://github.com/raiden-network/raiden/blob/325f4327583dc8e75677751865e63dda600bd929/raiden/transfer/mediated_transfer/target.py#L267) and clear it. 

So in conclusion what he is trying to say there is that if we did not return the target task we would never be able to handle the expired lock in the future.